### ISSUE TITLE
Installation script exits with status code zero upon missing dependencies

### ISSUE THREAD

**Issue by ddcprg** [sdkman/sdkman-cli#709]
- [] this is a new Feature Request, a conversation was started in the [Slack CLI Development Channel](https://sdkman.slack.com/app_redirect?channel=cli-development).
- [X] this is a new Bug Report or Issue, a conversation was started in [Slack User Issues Channel](https://sdkman.slack.com/app_redirect?channel=user-issues).

Installation scripts exits with 0 status code if a dependency is missing. This makes scripting the installation hard.

**Comment by Umofomia** [comment]
Just chiming in here to say that I've also noticed this issue. One time we had a momentary network error which caused the ZIP file to not download correctly. Because the installation script exited with a 0 status code, our Docker build cached the result of this as a successfully built image. This caused errors in downstream task that were difficult to debug until we realized that we needed to invalidate the bad image.
